packages/flutter/lib/src/widgets/widget_span.dart

```diff
@@ -90,17 +90,26 @@ class WidgetSpan extends PlaceholderSpan {
   /// Helper function for extracting [WidgetSpan]s in preorder, from the given
   /// [InlineSpan] as a list of widgets.
   ///
-  /// The `textScaleFactor` is the the number of font pixels for each logical
-  /// pixel.
+  /// The `textScaler` is the scaling strategy for scaling the content.
   ///
   /// This function is used by [EditableText] and [RichText] so calling it
   /// directly is rarely necessary.
-  static List<Widget> extractFromInlineSpan(InlineSpan span, double textScaleFactor) {
+  static List<Widget> extractFromInlineSpan(InlineSpan span, TextScaler textScaler) {
     final List<Widget> widgets = <Widget>[];
+    final List<double> fontSizeStack = <double>[14.0];
```

Comment:
Where's the magic 14.0 coming from?

Reply:
That's the default font size the engine uses if the TextSpan tree doesn't specify the font size. Added a comment.